securedrop_export/export.py

```diff
@@ -164,76 +164,84 @@ def extract_tarball(self):
-            msg = "USB_CONNECTED"
-            self.exit_gracefully(msg)
-        # Else the result is unexpected
-        else:
-            msg = "ERROR_USB_CHECK"
+            self.exit_gracefully("USB_NOT_CONNECTED")
+
+    def set_extracted_device_name(self):
+        try:
+            device_and_partitions = subprocess.check_output(
+                ["lsblk", "-o", "TYPE", "--noheadings", DEVICE], stderr=subprocess.PIPE)
+
+            # we don't support multiple partitions
+            partition_count = device_and_partitions.decode('utf-8').split('\n').count('part')
+            if partition_count > 1:
+                logging.debug("multiple partitions not supported")
+                self.exit_gracefully("USB_NO_SUPPORTED_ENCRYPTION")
+
+            # set device to /dev/sda if disk is encrypted, /dev/sda1 if partition encrypted
+            self.device = DEVICE if partition_count == 0 else DEVICE + '1'
+        except subprocess.CalledProcessError:
+            msg = "USB_NO_SUPPORTED_ENCRYPTION"
             self.exit_gracefully(msg)
 
     def check_luks_volume(self):
         logging.info('Checking if volume is luks-encrypted')
         try:
-            # cryptsetup isLuks returns 0 if the device is a luks volume
-            # subprocess with throw if the device is not luks (rc !=0)
-            subprocess.check_call(["sudo", "cryptsetup", "isLuks", DEVICE])
-            msg = "USB_ENCRYPTED"
-            self.exit_gracefully(msg)
+            self.set_extracted_device_name()
+            logging.debug("checking if {} is luks encrypted".format(self.device))
+            subprocess.check_call(["sudo", "cryptsetup", "isLuks", self.device])
+            self.exit_gracefully("USB_ENCRYPTED")
         except subprocess.CalledProcessError:
             msg = "USB_NO_SUPPORTED_ENCRYPTION"
             self.exit_gracefully(msg)
 
     def unlock_luks_volume(self, encryption_key):
-        # the luks device is not already unlocked
-        logging.info('Unlocking luks volume {}'.format(self.encrypted_device))
-        if not os.path.exists(os.path.join("/dev/mapper/", self.encrypted_device)):
-            p = subprocess.Popen(
-                ["sudo", "cryptsetup", "luksOpen", self.device, self.encrypted_device],
-                stdin=subprocess.PIPE,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE
-            )
-            logging.info('Passing key')
-            p.communicate(input=str.encode(encryption_key, "utf-8"))
-            rc = p.returncode
-            if rc != 0:
-                logging.error('Bad phassphrase for {}'.format(self.encrypted_device))
-                msg = "USB_BAD_PASSPHRASE"
-                self.exit_gracefully(msg)
+        try:
+            # get the encrypted device name
+            self.set_extracted_device_name()
+            luks_header = subprocess.check_output(["sudo", "cryptsetup", "luksDump", self.device])
```

Comment:
:+1: this will definitely improve the reliability of the luks checks. Since we shouldn't expect / support more than 1 luks volume this should work just fine.